lib/addons/argocd/index.ts

```diff
@@ -1,26 +1,359 @@
+    readonly options: ArgoCDAddOnProps;
+    private chartNode: HelmChart;
 
-    constructor(namespace?: string) {
-        this.namespace = namespace ?? "argocd";
+    constructor(props?: ArgoCDAddOnProps) {
+        this.options = { ...argoDefaults, ...props };
     }
 
-    deploy(clusterInfo: ClusterInfo): void {
-        clusterInfo.cluster.addHelmChart("argocd-addon", {
+    /**
+     * Impementation of the add-on contract deploy method.
+    */
+    async deploy(clusterInfo: ClusterInfo): Promise<Construct> {
+
+        const namespace = this.createNamespace(clusterInfo);
+
+        const sa = this.createServiceAccount(clusterInfo);
+        sa.node.addDependency(namespace);
+
+        const repo = await this.createSecretKey(clusterInfo, namespace);
+
+        const values = this.options.values ?? {
+            server: {
+                serviceAccount: {
+                    create: false
+                },
+                config: {
+                    repositories: repo
+                }
+            }
+        };
+
+        if(this.options.adminPasswordSecretName) {
+            const adminSecret = await this.createAdminSecret(clusterInfo.cluster.stack.region);
+            values['configs'] = {
+                secret: {
+                    argocdServerAdminPassword: adminSecret
+                }
+            }
+        }
+
+        this.chartNode = clusterInfo.cluster.addHelmChart("argocd-addon", {
             chart: "argo-cd",
-            
+            release: "ssp-addon",
             repository: "https://argoproj.github.io/argo-helm",
-            version: '3.2.3',
-            namespace: this.namespace,
+            version: '3.10.0',
+            namespace: this.options.namespace,
+            values: values
         });
+
+        this.chartNode.node.addDependency(sa);
+        return this.chartNode;
     }
 
-    postDeploy(clusterInfo: ClusterInfo, teams: Team[]): void {
-        //TODO: implement this method
-        console.assert(clusterInfo !== null && teams !== null);
+
+
+    /**
+     * Post deployment step is used to create a bootstrap repository if options are provided for the add-on.
+     * @param clusterInfo 
+     * @param teams 
+     * @returns 
+     */
+    async postDeploy(clusterInfo: ClusterInfo, teams: Team[]) {
+        console.assert(teams != null);
+        const appRepo = this.options.bootstrapRepo;
+
+        if (!appRepo) {
+            return;
+        }
+
+        const manifest = new KubernetesManifest(clusterInfo.cluster.stack, "bootstrap-app", {
+            cluster: clusterInfo.cluster,
+            manifest: [{
+                apiVersion: "argoproj.io/v1alpha1",
+                kind: "Application",
+                metadata: {
+                    name: appRepo.name ?? "bootstrap-apps",
+                    namespace: this.options.namespace
+                },
+                spec: {
+                    destination: {
+                        namespace: "default",
```

Comment:
Should this be `this.options.namespace`?

Reply:
Good catch. Apparently it does not affect anything for the app of apps, since there is no real deployable. Each app in the GitOps repo defines its own namespace, so everything works. Will fix.